scripts/create_test_subset.py

```diff
@@ -840,7 +845,40 @@ def get_random_families(
     return returned_families
 
 
-def copy_files_in_dict(d, dataset: str, sid_replacement: tuple[str, str] = None):
+def rewrite_file(old_path: str, new_path: str, new_base_path: str, sid: tuple[str, str]):
+    """
+    Returns the shell commands necessary to rewrite embedded IDs as appropriate
+    for each type of file likely to be encountered.
+    """
+    cmd: str | list[str]
+
+    if new_path.endswith('.cram'):
+        logger.info(f'Copying to {new_path} while rewriting @RG headers')
+        cmd = f"""
+        gcloud storage cat {old_path!r} |
+        samtools reheader --no-PG -c 'sed /^@RG/s/SM:{sid[0]}/SM:{sid[1]}/g' /dev/stdin |
+        gcloud storage cp - {new_path!r}
+        """
+    elif new_path.endswith('.crai'):
+        logger.info(f'Regenerating {new_path} by indexing {new_base_path}')
+        cmd = f"""
+        gcloud storage cat {new_base_path!r} | samtools index -o - - | gcloud storage cp - {new_path!r}
+        """
+    elif new_path.endswith('.md5'):
+        logger.info(f'Regenerating {new_path} by checksumming {new_base_path} [check the file format]')
+        cmd = f"""
+        gcloud storage cat {new_base_path!r} | md5sum | gcloud storage cp - {new_path!r}
+        """
```

Comment:
Can you replace this with a non-streaming
- copy CRAM
- reheader
- md5
- copy all results up
- rm local files

At the moment this implementation involves streaming the CRAM 3 times + 1 upload. The MD5 and `.crai` should be ignored during the copy as their contents are invalid post reheader

Reply:
That would be somewhat non-trivial given how the script is already organised, and would require local storage. This streaming is not egress, so two extra CRAM file reads from GCS vs complexity is a question of trade offs.

Reply:
It turns out that trying to run `samtools` commands directly from this script is a non-starter, because the script needs to run on an image that has the metamist API installed and any such images will not have samtools installed on them.

So this is now replaced by PR #1044, which creates a new Hail batch so is more at liberty to copy CRAMs around and do the work locally. Huzzah.